Decos.Diagnostics/ILog.cs

```diff
@@ -29,6 +31,32 @@ public interface ILog
         /// <param name="logLevel">The severity of the data.</param>
         /// <param name="data">The data to log.</param>
         void Write<T>(LogLevel logLevel, T data);
+
+        /// <summary>
+        /// Writes a message to the log with specified severity and customerID.
+        /// </summary>
+        /// <param name="logLevel">The severity of the message.</param>
+        /// <param name="message">The text of the message to log.</param>
+        /// <param name="customerID">The ID of the customer active on the moment of writing.</param>
+        void Write(LogLevel logLevel, string message, Guid customerID);
+
+        /// <summary>
+        /// Writes structured data to the log with the specified and customerID.
+        /// </summary>
+        /// <typeparam name="T">The type of data to write.</typeparam>
+        /// <param name="logLevel">The severity of the data.</param>
+        /// <param name="data">The data to log.</param>
+        /// <param name="customerID">The ID of the customer active on the moment of writing.</param>
+        void Write<T>(LogLevel logLevel, T data, Guid customerID);
+
+        /// <summary>
+        /// Writes structured data to the log with the specified data set in a LoggerContext object.
+        /// </summary>
+        /// <typeparam name="T">The type of data to write.</typeparam>
+        /// <param name="logLevel">The severity of the data.</param>
+        /// <param name="data">The data to log.</param>
+        /// <param name="senderDetails">Object containing data of the sender.</param>
+        void Write<T>(LogLevel logLevel, T data, LoggerContext senderDetails);
```

Comment:
Changing the interface will break existing implementations. It's better to add these as extension methods, if all you do is call other Write methods on the same instance.

Reply:
Changes made https://github.com/DecosInformationSolutions/Diagnostics/pull/6/commits/ec92872aaa451db5f31da5d4151be4c57c5e4a94 . 
Though I'm inexperienced with extension methods so I'm not sure if I implemented them the right way?